src/Carbon/include/HIToolbox/TextInputSources.h

```diff
@@ -0,0 +1,19 @@
+#ifndef _Carbon_TextInputSources_H_
+#define _Carbon_TextInputSources_H_
+
+#ifndef __cplusplus
```

Comment:
I don't have any experience with `extern "C"`, but going off of [stack overflow](https://stackoverflow.com/a/16192947/5988706), it should be `#ifdef __cplusplus` not `#ifndef __cplusplus`

However, I haven't tested this, so I could be wrong.